mlir/include/Ion/Transforms/oqd_database_managers.hpp

```diff
@@ -0,0 +1,206 @@
+                            "supported for decomposition onto beams.");
+        }
+
+        for (size_t i = 0; i < numBeams; i++) {
+            auto beam = beamsToml[i];
+            double rabi = beam["rabi"].as_floating_point()->get();
+            double detuning = beam["detuning"].as_floating_point()->get();
+            std::vector<int64_t> polarization =
+                tomlArray2StdVector<int64_t>(*(beam["polarization"].as_array()));
+            std::vector<int64_t> wavevector =
+                tomlArray2StdVector<int64_t>(*(beam["wavevector"].as_array()));
+
+            collector->push_back(Beam(rabi, detuning, polarization, wavevector));
+        }
+    }
+
+    void loadPhononParams()
+    {
+        toml::node_view<toml::node> phononsToml = sourceTomlGateDecomposition["phonons"];
+        size_t numPhononModes = phononsToml.as_array()->size();
+
+        auto parseSingleDirection = [](auto direction) {
+            double energy = direction["energy"].as_floating_point()->get();
+            std::vector<int64_t> eigenvector =
+                tomlArray2StdVector<int64_t>(*(direction["eigenvector"].as_array()));
+            return Phonon(energy, eigenvector);
+        };
+
+        for (size_t i = 0; i < numPhononModes; i++) {
+            auto phononMode = phononsToml[i];
+
+            Phonon COM_x = parseSingleDirection(phononMode["COM_x"]);
+            Phonon COM_y = parseSingleDirection(phononMode["COM_y"]);
+            Phonon COM_z = parseSingleDirection(phononMode["COM_z"]);
+
+            phonons.push_back(PhononMode(COM_x, COM_y, COM_z));
+        }
+    }
+
+    void loadIonParams()
+    {
+        toml::node_view<toml::node> ionsToml = sourceTomlQubit["ions"];
+
+        auto parseSingleLevel = [](auto level) {
+            int64_t principal = level["principal"].as_integer()->get();
+
+            std::vector<std::string> properties{"spin",
+                                                "orbital",
+                                                "nuclear",
+                                                "spin_orbital",
+                                                "spin_orbital_nuclear",
+                                                "spin_orbital_nuclear_magnetization",
+                                                "energy"};
+            std::vector<double> propertiesData(properties.size());
+
+            std::transform(properties.begin(), properties.end(), propertiesData.begin(),
+                           [&level](const std::string &name) {
+                               return level[name].as_floating_point()->get();
+                           });
+
+            return Level(principal, propertiesData[0], propertiesData[1], propertiesData[2],
+                         propertiesData[3], propertiesData[4], propertiesData[5],
+                         propertiesData[6]);
+        };
+
+        auto parseSingleTransition = [](const auto &transition_table,
```

Comment:
I would still call it `transition`, i.e. without the `table`, since it is still one single transition entry. The `as_table` from the callsite is because a single transition has the table of properties (levels and einsteinA), but to me the name "transition table" sounds like a table of multiple transitions. What do you think?

Reply:
When I first read through this I understood `transition` to mean that it was an instance of the `Transition` class, which is not that case. Maybe `transition_table_entry`, or just `transition_entry` (for the sake of brevity) would be better. Distinguishing this variable from the `Transition` class would improve readability, I think.

Reply:
> When I first read through this I understood `transition` to mean that it was an instance of the `Transition` class

Ahh, I see it now. I will change to `transition_entry`.